rest/blip_sync.go

```diff
@@ -141,20 +139,41 @@ func (ctx *blipSyncContext) register(profile string, handlerFn func(*blipHandler
 			if response := rq.Response(); response != nil {
 				response.SetError("HTTP", status, msg)
 			}
-			base.LogTo("Sync", "%s %q   --> %d %s ... %s", rq, profile, status, msg, ctx.effectiveUsername)
+			ctx.LogTo("Sync", "%s %q   --> %d %s ... %s", rq, profile, status, msg, ctx.effectiveUsername)
 		} else {
-			base.LogTo("Sync+", "%s %q   --> OK ... %s", rq, profile, ctx.effectiveUsername)
+			ctx.LogTo("Sync+", "%s %q   --> OK ... %s", rq, profile, ctx.effectiveUsername)
 		}
 	}
 }
 
 // Handler for unknown requests
 func (ctx *blipSyncContext) notFound(rq *blip.Message) {
-	base.LogTo("Sync", "%s %q ... %s", rq, rq.Profile(), ctx.effectiveUsername)
-	base.LogTo("Sync", "%s    --> 404 Unknown profile ... %s", rq, ctx.effectiveUsername)
+	ctx.LogTo("Sync", "%s %q ... %s", rq, rq.Profile(), ctx.effectiveUsername)
+	ctx.LogTo("Sync", "%s    --> 404 Unknown profile ... %s", rq, ctx.effectiveUsername)
 	blip.Unhandled(rq)
 }
 
+// Prepend a context ID to each blip logging message.  The contextID uniquely identifies the blip context, and
+// is useful for grouping the blip connections in the log output.
+func PrependContextID(contextID, format string, params ...interface{}) (newFormat string, newParams []interface{}) {
+
+	// Add a new format placeholder for the context ID, which should appear at the beginning of the logs
+	formatWithContextID := `[%s] ` + format
+
+	paramsWithContextID := []interface{}{contextID}
+	if len(params) > 0 {
+		paramsWithContextID = append(paramsWithContextID, params...)
+	}
```

Comment:
Can this use the standard slice insert approach to avoid allocating the new slice? (from https://github.com/golang/go/wiki/SliceTricks)

Reply:
Fixed